src/Tools/dotnet-monitor/TestAssemblies.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+using System;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Hosting;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    /// <summary>
+    /// Methods for loading test assemblies into the tool when running from the build output of the repository.
+    /// </summary>
+    internal static class TestAssemblies
+    {
+        private const string ArtifactsDirectoryName = "artifacts";
+        private const string TestHostingStartupAssemblyName = "Microsoft.Diagnostics.Monitoring.Tool.TestHostingStartup";
+        private const string TestStartupHookAssemblyName = "Microsoft.Diagnostics.Monitoring.Tool.TestStartupHook";
+
+        [Conditional("DEBUG")]
+        public static void SimulateStartupHook()
+        {
+            // This code is to aid loading the TestStartupHook assembly when debug launching dotnet-monitor
+            // so that additional manual configuration is not necessary. The functional tests already bootstrap
+            // loading this assembly and initialize the hosting startup using standard dotnet environment variables.
+
+            AssemblyLoadContext alc = AssemblyLoadContext.GetLoadContext(Assembly.GetExecutingAssembly());
+            // Skip attempting to load if it is already loaded
+            if (!alc.Assemblies.Any(a => TestStartupHookAssemblyName.Equals(a.GetName().Name)))
+            {
+                // Only load if can compute the path correctly
+                if (TryComputeBuildOutputAssemblyPath(TestStartupHookAssemblyName, out string startupHookAssemblyPath))
+                {
+                    // Load and simulate startup hook initialization
+                    Assembly startupHookAssembly = alc.LoadFromAssemblyPath(startupHookAssemblyPath);
+                    Type startupHookType = startupHookAssembly.GetType("StartupHook");
```

Comment:
nit: I'm wondering if we can do less direct binding to these names so that renaming won't just break this at runtime. For example, using an assembly level attribute that tells dotnet-monitor which type to look for.

Reply:
The startup hook notion is a public contract where a type named `StartupHook` has a static `Initialize` method. This is highly unlikely to ever change. See [runtime](https://github.com/dotnet/runtime/blob/6246f84506a7f0a2cf990c2fddd7225691f4bc89/src/coreclr/System.Private.CoreLib/src/System/StartupHookProvider.cs#L114) startup hook processing code, which the code that I wrote is a highly simplified version of it.